src/data/set/intervals/monotone.lean

```diff
@@ -177,3 +178,63 @@ calc (⋃ x, Ioo (f x) (g x)) = (⋃ x, Ioi (f x)) ∩ ⋃ x, Iio (g x) :
 ... = Ioi a ∩ Iio b : congr_arg2 (∩) ha.Union_Ioi_eq hb.Union_Iio_eq
 
 end Union
+
+section succ_order
+
+open order
+
+variables {α β : Type*} [partial_order α] [succ_order α] [is_succ_archimedean α]
+
+lemma strict_mono_on.Iic_id_le [order_bot α] {n : α} {φ : α → α}
+  (hφ : strict_mono_on φ (set.Iic n)) :
```

Comment:
Is there a corresponding lemma about `pred_order` and `strict_anti_on`? If not, can you add a brief comment as to why?